src/serac/infrastructure/input.cpp

```diff
@@ -67,24 +67,47 @@ std::string fullDirectoryFromPath(const std::string& path)
   return dir;
 }
 
-void defineVectorInputFileSchema(axom::inlet::Table& table, const int dimension)
+void defineVectorInputFileSchema(axom::inlet::Table& table)
 {
-  if (dimension < 0 || dimension > 3) {
-    SLIC_ERROR("Cannot define an input file schema for vector of invalid size" << dimension);
-  }
-  table.addDouble("x", "x-component of vector").required(true);
-  if (dimension >= 2) {
-    table.addDouble("y", "y-component of vector").required(true);
-    if (dimension >= 3) {
-      table.addDouble("z", "z-component of vector").required(true);
-    }
-  }
+  table.addDouble("x", "x-component of vector").required();
+  table.addDouble("y", "y-component of vector");
+  table.addDouble("z", "z-component of vector");
 }
 
 void BoundaryConditionInputOptions::defineInputFileSchema(axom::inlet::Table& table)
 {
-  table.addString("name", "Name used to identify the boundary condition").required();
   table.addIntArray("attrs", "Boundary attributes to which the BC should be applied").required();
+  CoefficientInputOptions::defineInputFileSchema(table);
+}
+
+bool CoefficientInputOptions::isVector() const
+{
+  return std::holds_alternative<CoefficientInputOptions::VecFunc>(func);
+}
+
+mfem::VectorFunctionCoefficient CoefficientInputOptions::constructVector(const int dim) const
+{
+  auto vec_func = std::get_if<CoefficientInputOptions::VecFunc>(&func);
+  SLIC_ERROR_IF(!vec_func, "Cannot construct a vector coefficient from a scalar function");
+  return {dim, *vec_func};
+}
+
+mfem::FunctionCoefficient CoefficientInputOptions::constructScalar() const
+{
+  auto scalar_func = std::get_if<CoefficientInputOptions::ScalarFunc>(&func);
+  SLIC_ERROR_IF(!scalar_func, "Cannot construct a scalar coefficient from a vector function");
+  return {*scalar_func};
+}
+
+void CoefficientInputOptions::defineInputFileSchema(axom::inlet::Table& table)
+{
+  table.addFunction("vec_coef", axom::inlet::FunctionType::Vec3D,
+                    {axom::inlet::FunctionType::Vec3D},  // Multiple argument types
```

Comment:
are you explaining this converts automatically to x,y,z? might be better if the comment says:

`// Automatically converted to x, y, z in Lua